tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip(' \n')
+        if container_state in ["enabled", "disabled"]:
+           autorestart_containers[container_name] = container_state
+
+    return autorestart_containers
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled by default
+    @return: A list includes the names of disabled container/service
+    """
+    disabled_containers = []
+    container_list = []
+
+    redis_cli_keys_output = duthost.shell("redis-cli -n 4 keys 'FEATURE|*'")
+    for line in redis_cli_keys_output["stdout_lines"]:
+        container_list.append(line.strip(' \n').split('|')[1])
+
+    for container_name in container_list:
+        redis_cli_values_output = duthost.shell("redis-cli -n 4 hgetall 'FEATURE|{}'".format(container_name))
+        values = redis_cli_values_output["stdout_lines"]
+        if "status" in values and (values.index("status") + 1 < len(values)) \
+                and values[values.index("status") + 1] == "disabled":
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+
+    """
+    is_running = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    return is_running["stdout_lines"][0].strip(' \n')
+
+def get_program_status(duthost, container_name, program_name):
+    """
+    @summary: Return the status of a program in the specified container
+    @return:  "RUNNING" or "EXITED" represents the program is in the running
+              or exited status
+    """
+    program_status = None
+
+    process_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
+    for process_info in process_list["stdout_lines"]:
+        if process_info.find(program_name) != -1:
+            program_status = process_info.split()[1].strip()
+            break
+ 
+    return program_status
+
+def kill_process_by_pid(duthost, container_name, program_name, program_status, program_pid):
+    """
+    @summary: Kill a process in the specified container by its pid
+
+    """
+    if program_status == "RUNNING":
+        duthost.shell("docker exec {} kill -SIGKILL {}".format(container_name, program_pid)) 
+    elif program_status in ["EXITED", "STOPPED", "STARTING"]:
+        pytest_assert(False, "{} in {} is in the {} state not in the running state" \
+                .format(program_name, container_name, program_status))
+    else:
+        pytest_assert(False, "Failed to find {} in {} container" \
+                .format(program_name, container_name))
+    # Sleep 10 seconds to wait container is stopped
+    time.sleep(10)
```

Comment:
Instead of a hard 10 second sleep here for every container, can you use `wait_until()` to wait for the service to be stopped? This could help speed the test up if most containers stop in less than 10 seconds

Reply:
Great suggestion! Will update.